# Question 1
The following is a list of 10 students ages:
ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
• Sort the list and find the min and max age
• Add the min age and the max age again to the list
• Find the median age (one middle item or two middle items divided by two)
• Find the average age (sum of all items divided by their number)
• Find the range of the ages (max minus min)

In [ ]:
ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
ages.sort()
print("Sorted Ages", ages)
print("min age",min(ages))
print("max age",max(ages))
ages.append(min(ages))
ages.append(max(ages))
print("Added the min & max age again",ages) 
#Median Age using statistics library
import statistics
print("Median", statistics.median(ages))
print("average age" ,sum(ages)/len(ages))
print("Range" ,max(ages)-min(ages))


# Question 2
• Create an empty dictionary called dog
• Add name, color, breed, legs, age to the dog dictionary
• Create a student dictionary and add first_name, last_name, gender, age, marital status,
skills, country, city and address as keys for the dictionary
• Get the length of the student dictionary
• Get the value of skills and check the data type, it should be a list
• Modify the skills values by adding one or two skills
• Get the dictionary keys as a list
• Get the dictionary values as a list

In [ ]:
dog={}
dog['name']='value1'
dog['color']='value2'
dog['breed']='value3'
dog['legs']='value4'
dog['age']='value5'
print(dog)

student={}
student['first_name']='value1'
student['last_name']='value2'
student['gender']='value3'
student['age']='value4'
student['marital status']='value5'
student['skills']=['JavaScript', 'React', 'Node', 'MongoDB', 'Python']
student['country']='value7'
student['city']='value8'
student['address']='value9'
print(student)
print("length of the student dictionary: ",len(student))
print(type(student['skills']))
student['skills'].append('C++')

print(student.keys())
print(student.values())

# Question 3
• Create a tuple containing names of your sisters and your brothers (imaginary siblings are
fine)
• Join brothers and sisters tuples and assign it to siblings
• How many siblings do you have?
• Modify the siblings tuple and add the name of your father and mother and assign it to
family_members


In [ ]:
sisters=('tehmina','maliha','ayesha')
brothers=('khalid','salman','riyyan')
siblings= sisters+brothers

print(siblings)
print(len(siblings))

family_members= siblings + ('Jeelani','Rehana')
print(family_members)

# Question 4
it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}
age = [22, 19, 24, 25, 26, 24, 25, 24]
• Find the length of the set it_companies
• Add 'Twitter' to it_companies
• Insert multiple IT companies at once to the set it_companies
• Remove one of the companies from the set it_companies
• What is the difference between remove and discard
• Join A and B
• Find A intersection B
• Is A subset of B
• Are A and B disjoint sets
• Join A with B and B with A
• What is the symmetric difference between A and B
• Delete the sets completely
• Convert the ages to a set and compare the length of the list and the set.

In [ ]:
it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}


print("length of the set it_companies", len(it_companies))
it_companies.add("Twitter")
print(it_companies)

it_companies.update(['Issquared','Cognizent','Accenture'])
print(it_companies)

it_companies.remove('Cognizent') #Remove eliminates a particular given item from a set
print(it_companies)

it_companies.clear() #Clears all the items in the set 
print(it_companies)

del it_companies #discards the entire set
print(it_companies)

• Join A and B.
• Find A intersection B.
• Is A subset of B.
• Are A and B disjoint sets.
• Join A with B and B with A
• What is the symmetric difference between A and B.
• Delete the sets completely.
• Convert the ages to a set and compare the length of the list and the set.

In [1]:
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}

print(A.union(B))
print(A.intersection(B))
print(A.issubset(B))
print(A.isdisjoint(B))

print(A.union(B))
print(B.union(A))
print(A.symmetric_difference(B))

del A
del B
print(A)
print(B)

{19, 20, 22, 24, 25, 26, 27, 28}
{19, 20, 22, 24, 25, 26}
True
False
{19, 20, 22, 24, 25, 26, 27, 28}
{19, 20, 22, 24, 25, 26, 27, 28}
{27, 28}


NameError: name 'A' is not defined

In [2]:
age = [22, 19, 24, 25, 26, 24, 25, 24]
print(set(age))
print(len(age)-len(set(age)))

{19, 22, 24, 25, 26}
3


# Question 5
The radius of a circle is 30 meters.
• Calculate the area of a circle and assign the value to a variable name of _area_of_circle_
• Calculate the circumference of a circle and assign the value to a variable name of
_circum_of_circle_
• Take radius as user input and calculate the area.

In [3]:
r=30
_area_of_circle_ = 3.14*(r**2)
print(_area_of_circle_)

_circum_of_circle_= 2*3.14*r 
print(_circum_of_circle_)

R=int(input("enter radius value to calculate Area of circle"))
_area= 3.14*R**2
print(_area)

2826.0
188.4
enter radius value to calculate Area of circle10
314.0


# Question 6
“I am a teacher and I love to inspire and teach people”
• How many unique words have been used in the sentence? Use the split methods and set
to get the unique words.

In [ ]:
string = 'I am a teacher and I love to inspire and teach people'
print(len(set(string.split())))

# Question 7
Use a tab escape sequence to get the following lines.
 Name Age Country City
 Asabeneh 250 Finland Helsinki

In [ ]:
print('Name\t\t Age\t Country\t City')
print('Asabeneh\t 250\t Finland\t Helsinki')

In [11]:
print('Name\t\t Age\t Country\t City \nAsabeneh\t 250\t Finland\t Helsinki')


Name		 Age	 Country	 City 
Asabeneh	 250	 Finland	 Helsinki


# Question 8
Use the string formatting method to display the following:
radius = 10
area = 3.14 * radius ** 2
“The area of a circle with radius 10 is 314 meters square.”

In [ ]:
radius = 10
area = 3.14 * radius ** 2
print('The area of a circle with radius {} is {} meters square.'.format(str(radius),str(area)))

# Question 9
Write a program, which reads weights (lbs.) of N students into a list and convert these weights to
kilograms in a separate list using Loop. N: No of students (Read input from user)
Ex: L1: [150, 155, 145, 148]
Output: [68.03, 70.3, 65.77, 67.13]

In [2]:
student_count= int(input("Enter number of students "))

weight_in_lb = list(map(int,input("\nEnter the student weights: ").strip().split()))[:student_count]

weight_in_kg = []

for i in weight_in_lb:
    
    kg = i * 0.45
    weight_in_kg.append(kg)
    
    
print('The weights of students in kgs is',weight_in_kg)

Enter number of students 4

Enter the student weights: 120 140 199 150
The weights of students in kgs is [54.0, 63.0, 89.55, 67.5]
